wit/guest.wit

```diff
@@ -3,8 +3,7 @@ interface incoming-handler {
 
```

Comment:
You deleted the `consumer.wit`  interface, but I do think we should keep the `subscribe` method. Otherwise, how will `handle` know from where to receive messages from? Maybe make `subscribe` part of this interface with the intention of it being an `export`, just like `handle`. My mental model right now is that a Wasm host would call a guest's `subscribe` and then use `handle` for whatever is sent by a `producer` to the topics returned from `subscribe`.

Reply:
I am not opposed to adding it back in down the line, but I think the subscriptions get defined at component start time and not by the component. It eliminates the potential of a security issue (untrusted code being able to subscribe to something it shouldn't). This is a common pattern across things. wasmCloud defines these via configuration at link time, Spin defines this in their spin.toml file, etc. Essentially configuration doesn't come from the component, it comes from the person running the component. I do know there are some use cases where something might subscribe to more topics based on a received message, but I wanted to pare it back to what I know works based off of users experience with wasmcloud:messaging. 

Doing it this way allows us to add it with a `@since` directive in a subsequent version without too much extra hassle while modeling it around an actual incoming use case.

Reply:
Yeah, I figured the resposibility of subscriptions would be left to some sort of config file. Feels a bit weird for the interface to depend on something external like that for it to work, but I understand the security benefits you highlighted. That said, being able to subscribe to topics at runtime still feels like a must have to me. If other champions are good with this design, I'm not going to block on this and we can move forward and maybe just leverage `@since` down the line.